channeldb/payment_control.go

```diff
@@ -360,14 +360,7 @@ func (p *PaymentControl) RegisterAttempt(paymentHash lntypes.Hash,
 			return err
 		}
 
-		// Create bucket for this attempt. Fail if the bucket already
-		// exists.
-		htlcBucket, err := htlcsBucket.CreateBucket(htlcIDBytes)
-		if err != nil {
-			return err
-		}
-
-		err = htlcBucket.Put(htlcAttemptInfoKey, htlcInfoBytes)
+		err = htlcsBucket.Put(append(htlcAttemptInfoKey, htlcIDBytes...), htlcInfoBytes)
```

Comment:
We need to be careful when using `append()` with global variables as sometimes that can modify their value. See https://trstringer.com/golang-append/

Reply:
> Great work on this, looks like you were able to identify and address quite a number of bottlenecks.
> I can't say I fully understand the whole code around STM just yet, so take my review with a grain of salt. But the changes look good as far as I can tell.

Thanks for the review @guggero !! :)